tests/common/configuration/node_config_model.rs

```diff
@@ -89,7 +89,7 @@ impl NodeConfig {
 
 fn get_available_port() -> u16 {
     let available_port = loop {
-        let port = rand::thread_rng().gen_range(8000, 9999);
+        let port = rand::thread_rng().gen_range(6000, 9999);
```

Comment:
I don't think that this change will fix the test unstability regarding port binding. This loop never fails and always returns a free port. If lack of ports was problem, it would hang forever and not panic.

I suspect that the real cause is that 2 threads generated the same port, validated it at different times, both succesfuly, then one started using it and the other got rejected. The time gap between validation and usage creates a race condition.

Reply:
What I think we could do here is use the fact that every test binary is run sequentially in separate process and only tests functions are run in parallell threads. If we have a static (atomic!) counter with last used port, we could eliminate conflicts. Every process will then create it from a scratch with value 8000 (no conflict here, processes are run sequentially). During search for a free port it could be incremented on every try in every thread. We probably won't run to 9999 and if we do, we probably have a problem anyway and should panic anyway.

Reply:
Could port number ranges be assigned by the test runner, so that they could not conflict between processes?

Reply:
I'm not aware of such tool. Isn't it enough that processes don't run in parallel?